src/comparator/field_comparator.py

```diff
@@ -232,19 +266,21 @@ def _resource_ref_in_local(self, resource_ref):
             raise TypeError(
                 "In a resource_reference annotation, either `type` or `child_type` field should be defined"
             )
-        if self.local_resource_update.value.type != checked_type:
+        if self.field_original.child_type:
+            parent_resources = (
+                self.global_resources_update.get_parent_resources_by_child_type(
+                    resource_ref.child_type
+                )
+            )
+            if not any(
+                self.local_resource_update.value.type == resource.value.type
+                for resource in parent_resources
+            ):
+                return False
+        elif self.local_resource_update.value.type != resource_ref.type:
             return False
         return True
 
-    def _register_local_resource(self):
```

Comment:
The local resource (message options) is already registered in the database, so no need to add again.